sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroSource.java

```diff
@@ -270,20 +276,19 @@ public void validate() {
       }
       codec = metadata.getCodec();
       syncMarker = metadata.getSyncMarker();
-      fileSchemaString = metadata.getSchemaString();
-      // If the source was created with a null schema, use the schema that we read from the file's
-      // metadata.
-      if (readSchemaString == null) {
-        readSchemaString = metadata.getSchemaString();
+
+      // Only store both the read and file schema if they differ
```

Comment:
Yes, this is to save memory and is only important in the case of initial splitting. If a schema is 100KiB and you have 10,000 splits, then you would have needed at least 1 GiB of memory to store all the copies.

I'm confused about your point to store a reference since all objects are already references, are you speaking about the java Reference type?